src/classes/groundplaces.ts

```diff
@@ -201,31 +194,119 @@ export class GroundPlacesController {
+      );
+    }
+
+    // Remove the reference of the StopGroup Gpuid inside the StopCluster parent
+    stopClusterParent.childs.splice(stopGroupIndex, 1);
+
+    // Update the place with the new StopCluster parent
+    this.storageService.replacePlace(stopClusterParent);
+
+    // Check if the StopGroup is not lonely after the operation
+    const stopGroupExist = groundPlaces.find(
+      (groundPlace: GroundPlace): boolean =>
+        groundPlace.type === GroundPlaceType.CLUSTER &&
+        Boolean(groundPlace.childs.find((stopGroupGpuid) => stopGroupGpuid === stopGroupGpuidToRemove)),
+    );
+
+    // If the StopGroup is lonely, rollback to the previous version of the ground places stored
+    if (!stopGroupExist) {
+      this.storageService.setGroundPlaces(copyGroundPlaces);
+
+      throw new Error(
+        `Impossible to remove the StopGroup with the Gpuid "${stopGroupGpuidToRemove}". Make sure that the StopGroup you want to remove will not be without any StopCluster parent after this operation.`,
+      );
+    }
+  }
+
   /**
    * @description Move a stopGroup from a stopCluster to another stopCluster.
    * @param {StopGroupGpuid} stopGroupToMoveGpuid - Ground place unique identifier of the stopGroup to move.
    * @param {StopClusterGpuid} fromStopClusterGpuid - Ground place unique identifier of the old stopCluster.
    * @param {StopClusterGpuid} intoStopClusterGpuid - Ground place unique identifier of the new stopCluster.
-   * @returns {StopCluster}
+   * @returns {void}
    */
   public moveStopGroup(
     stopGroupToMoveGpuid: StopGroupGpuid,
     fromStopClusterGpuid: StopClusterGpuid,
     intoStopClusterGpuid: StopClusterGpuid,
-  ): void {}
+  ): void {
+    if (fromStopClusterGpuid === intoStopClusterGpuid) {
+      throw new Error(
+        `You can't move the StopGroup with the Gpuid "${stopGroupToMoveGpuid}" because it already exists inside the new StopCluster parent specified.`,
+      );
+    }
 
-  /**
-   * @description Add a stopGroup to a stopCluster.
-   * @param {StopGroupGpuid} stopGroupToAddGpuid - Ground place unique identifier of the stopGroup to add.
-   * @param {StopClusterGpuid} intoStopClusterGpuid - Ground Place unique identifier of the stopCluster.
-   * @returns {void}
-   */
-  public addStopGroupToStopCluster(stopGroupToAddGpuid: StopGroupGpuid, intoStopClusterGpuid: StopClusterGpuid): void {}
+    const copyGroundPlaces: GroundPlace[] = this.storageService.cloneGroundPlaces();
+
+    try {
+      this.addStopGroupToStopCluster(stopGroupToMoveGpuid, intoStopClusterGpuid);
+      this.removeStopGroupFromStopCluster(stopGroupToMoveGpuid, fromStopClusterGpuid);
+    } catch (error) {
+      // If there is error in the process, rollback to the previous version of the ground places stored
+      this.storageService.setGroundPlaces(copyGroundPlaces);
+
+      throw new Error(error.message);
```

Comment:
why not throw error ?

Reply:
To avoid the double `Error` present in the output error message, ou can see on the screenshot above the Error.message on first line and the Error on the second line
![Capture d’écran 2020-12-11 à 14 41 54](https://user-images.githubusercontent.com/19270653/101910375-24601100-3bbf-11eb-88db-5611d352671c.png)

Reply:
ok